lib/sync_engine/2020_01_15/user_manager.rb

```diff
@@ -0,0 +1,39 @@
+module SyncEngine
+  module V20200115
+    class UserManager < SyncEngine::V20190520::UserManager
+      def change_pw(user, password, params)
+        current_protocol_version = user.version.to_i
+        new_protocol_version = params[:version].to_i || current_protocol_version
+
+        upgrading_protocol_version = new_protocol_version > current_protocol_version
+
+        user.encrypted_password = hash_password(password)
+        user.update!(registration_params(params))
+
+        # We want to create a new session only if upgrading from a protocol version that does not
+        # support sessions (i.e: 003 to 004).
+        if upgrading_protocol_version
+          handle_auth_response(user, params)
+        else
+          { user: user }
+        end
+      end
+
+      private
+
+      def handle_auth_response(user, params)
+        session = Session.new(user_uuid: user.uuid, api_version: params[:api], user_agent: params[:user_agent])
+
+        unless session.save
+          return { error: { message: 'Could not create a session.', status: 400 } }
+        end
+
+        response = session.response_hash
```

Comment:
`session.response_hash` returns the `session` and `token` keys. Although, I can also include the `user` key here and only return `session.response_hash`.

Reply:
I'm just saying generally that rather than creating an object and modifying it in the next line to add user, you should create the object as you want it once with all the info it needs, and return it directly.